use send_stat instead of overriding ruby send method

sensadrome opened the pull request:
in some implementations of the code we might want to call something like:

type = :increment

$statsd.send(type, my_stat)

but this would not work because send has been aliased to one of send_to_buffer or send_to_socket

yannmh commented:
Thanks a lot @sensadrome for your contribution.

I really like the code cleaning :smile:  but I am not sure to understand what's exactly the issue you're experiencing with `statsd.send`, and how your changes solve it :confounded: . Could you give us more details please ? 

As a side note, it seems that these changes break the test suite.

yannmh commented:
@sensadrome any update about this PR ?

sensadrome commented:
sorry - been busy! - so "send" is an inbuilt ruby method on Object (http://ruby-doc.org/core-2.2.2/Object.html#method-i-send) - technically speaking we could use __send__ instead which I think another dev suggested. In your code you specify "alias :send :send_to_socket" which effectively overwrites the Object.send method and thus leads to unexpected behaviour....

essentially your send method switches between send_to_socket and send_to_buffer which is nice - but instead of using :send as the alias I have used :send_stat

Hope that makes sense!

sensadrome commented:
OK - I have updated the code and adjusted the spec helpers to initialise the object with a hash instead of ordered args. Tests all pass now ... sorry, should have done that before!

sensadrome commented:
should ruby 1.8.7 be removed as a target build, since i18n requires 1.9.3 ?

yannmh commented on lib/statsd.rb:
```diff
@@ -60,36 +59,39 @@ def self.VERSION
-  def initialize(host = DEFAULT_HOST, port = DEFAULT_PORT, opts = {}, max_buffer_size=50)
-    self.host, self.port = host, port
-    @prefix = nil
+  def initialize(opts = {})
```
This is unfortunately breaking backward incompatibility :cry: , i.e.
```ruby
s = Statsd.new('localhost', 1234)
ArgumentError: wrong number of arguments (2 for 0..1)
```

yannmh commented:
Thanks a lot for your work and the further information @sensadrome !

We'd like to merge your contribution and release a new gem. Unfortunately some of them are breaking backward incompatibility (i.e. comment made above). 
If you don't mind, I'll restablish `initialize` method original signature, and also rebase your work on top of current master state (to solve merge conflicts). Please let me know what you think.

yannmh commented:
To avoid backward incompatibilities issues, I opened a new PR #17 with your first commit cherry-picked. 
I am also closing this one: I think it's wiser to bring other commits in a different PR, so they can be reviewed independently. Please feel free to re-open a new one :smiley: .

Thanks again.

sensadrome commented:
yes fair enough - I hadn't thought about backward compatibility. Happy to add the code to cover it if you like - right now we're just using the gem directly from our own repo, but ultimately would be better to use yours :)

Let me know...

yannmh commented:
@sensadrome, that would be great ! Thank you
